### ISSUE TITLE
We cannot undo detaching the child row from a parent

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#6723]
### Description

This issue is only replicable in `8.0.0-beta1` but only because it was never possible to detach a child. That issue has been described by @aninde here https://github.com/handsontable/handsontable/issues/6432

### Steps to reproduce

1. Go to https://jsfiddle.net/k1wzsce4/5/
2. Use `Detach from parent` on the `Alabama Shakes`
3. Use <kbd>CTRL/CMD</kbd> + <kbd>Z</kbd> (undo action)

**Result**
* we get a console error
![image](https://user-images.githubusercontent.com/10757813/75155085-84102d00-570f-11ea-927f-4c7cba903161.png)

If we hit the same button combination again

* the indexes are incorrect just like it is described here https://github.com/handsontable/handsontable/issues/6433
![image](https://user-images.githubusercontent.com/10757813/75155252-e9fcb480-570f-11ea-9d3d-7b16aa6194e8.png)

Replication video
![error](https://user-images.githubusercontent.com/10757813/75155365-30eaaa00-5710-11ea-98d8-8cde8a477bdf.gif)

### Additional info
* `beforeUndo` returns false results https://jsfiddle.net/k1wzsce4/7/
I'm detaching row index `1` and even if it is gone and would appear last - it is still index `3` not `4` 
![image](https://user-images.githubusercontent.com/10757813/75155776-05b48a80-5711-11ea-9400-a46e2f02a5fa.png)
If we add another child there's also 1 index mismatch 
![image](https://user-images.githubusercontent.com/10757813/75155901-47ddcc00-5711-11ea-92bf-e573dd695b5c.png)

* `afterUndo` doesn't fire https://jsfiddle.net/k1wzsce4/6/

### Related issues
* https://github.com/handsontable/handsontable/issues/6433
* https://github.com/handsontable/handsontable/issues/6432

### Demo

https://jsfiddle.net/handsoncode/8ffpsqt6/ (`8.0.0-beta1`)

### Your environment
* Handsontable version: 8.0.0-beta1
* Browser Name and version: Chrom 80
* Operating System: Windows 10

**Comment by aninde** [comment]
Same error in console: _Cannot read property "__children" of null_ was reported in bug with nested rows with loadData https://github.com/handsontable/handsontable/issues/3914#issuecomment-470844376

**Comment by AMBudnik** [comment]
I'm glad to see that this issue is **solved** with 8.0.0.-beta2-rev2 https://jsfiddle.net/1jo96z8u/ 

![hurrra](https://user-images.githubusercontent.com/10757813/81558060-4ecdbf80-938d-11ea-8d70-d46f9156bacb.gif)

**Comment by aninde** [comment]
Sadly this issue is still here, but with different error message. This bug describes undoing detaching, not removing. 
I was able to reproduce it also in @AMBudnik  demo in `rev4` https://github.com/handsontable/handsontable/issues/6723#issuecomment-626651402
<img width="684" alt="Screenshot 2020-05-15 at 10 58 18" src="https://user-images.githubusercontent.com/30186818/82032175-28649880-969b-11ea-9806-5065d9cbaf20.png">

![undo-detachprebeta](https://user-images.githubusercontent.com/30186818/82031797-a7a59c80-969a-11ea-970b-763742d5a025.gif)

**Reproduction:**
1. Detach one nested row from parent via `contextMenu`
2. Undo it by `cmd + d`

**Demo**
https://jsfiddle.net/aninde/rhbm8f07/ `pre8.0.0-beta2-rev1` we got detached row, but there is an error in console `script error`

 https://jsfiddle.net/aninde/rhbm8f07/ `pre8.0.0-beta2-rev6`, we got detached row, but there is an error in console `script error`
https://jsfiddle.net/aninde/vo042gjt/ `v7.4.2`  - broken even worse, we lost detached row, `script error` in console

**Comment by AMBudnik** [comment]
You're right - I took 'remove' not 'detach'. I'm taking off the Releases.

**Comment by aninde** [comment]
Updated dome with `v8.0.0` **the behavior has changed**. https://jsfiddle.net/aninde/j45ksma0/ but it is still broken.

- there is no longer an error in the console
- **the action is different after undo**  `Alabama shakes` renders where it should be `Florence..`
- after the second undo we receive two `Alabama shakes`. `Florence..` is lost. 
<img width="582" alt="Screenshot 2020-08-12 at 13 37 28" src="https://user-images.githubusercontent.com/30186818/90011081-442f0900-dca1-11ea-82f6-917a034fb0f8.png">






![undo-detach](https://user-images.githubusercontent.com/30186818/90010458-2a40f680-dca0-11ea-863a-8cc38921d78b.gif)

Improper result after two undo:
<img width="586" alt="Screenshot 2020-08-12 at 13 31 53" src="https://user-images.githubusercontent.com/30186818/90010498-388f1280-dca0-11ea-87a1-f1e1643c395e.png">
